### PR TITLE
Change display of friends to single row + some refactoring of "footer"

### PR DESCRIPTION
**Pull request by yipingliao**
Limit friends display in module in right column to just one row (4 friends).

Asana task: https://app.asana.com/0/15014070792041/15100933478456
A screenshot of what it looks like in dev is included in the task.

Also, the parent Asana task has a mockup image: https://app.asana.com/0/15014070792041/13515857251339

Finally, a little bit of refactoring in renaming "footer" to "rightCol".

@2is10 @andrewconner

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/app.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ angular.module('kifi', [
   'kifi.friendService',
   'kifi.friends.friendCard',
   'kifi.friends.friendRequestCard',
-  'kifi.friends.compactFriendsView',
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/app.js:26]: This module now contains directives for both 'compactFriendsView' as well as 'noFriendsOrConnectionsView', so I am renaming it to 'rightColFriendsView'.

File: kifi-backend/modules/shoebox/angular/src/friends/compactFriendsView.styl [hunk 2]
```diff
@@ -1,13 +1,11 @@
 .kf-compact-friend-image
-  height: 40px
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/friends/compactFriendsView.styl:2]: Nothing changed here for .kf-compact-friend-image; alpha-sorted properties for quicker scanning.

Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/friends/compactFriendsView.styl:2]: That’s an interesting convention. Mine has been to work from the outside in: positioning (e.g. `display`, `position`, `float`), size, margin, border, background, padding, text alignment, text formatting. This way related properties, like width & height, stay together even though they aren’t alphabetically close together. I don’t think we’ve ever discussed property ordering conventions as a team, but I think discussing it as a group is a great idea.

File: kifi-backend/modules/shoebox/angular/src/friends/rightColFriendsView.js [hunk 3]
```diff
@@ -9,23 +9,37 @@ angular.module('kifi.friends.compactFriendsView', [])
...
+      scope.friendsDisplayLimit = 4;
+
+      // Get number of friends.
       scope.friendCount = friendService.totalFriends;
+
+      // Populate local friends list.
       friendService.getKifiFriends().then(function (data) {
         var actualFriends = _.filter(data, function (friend) {
+          // Attach friend's picture url to local friends list.
           friend.pictureUrl = friendService.getPictureUrlForUser(friend);
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/friends/rightColFriendsView.js]: This line is unrelated to the filtering going on. Probably clearer to move it to its own `forEach`.

File: kifi-backend/modules/shoebox/angular/src/friends/rightColFriendsView.js [hunk 4]
```diff
@@ -9,23 +9,37 @@ angular.module('kifi.friends.compactFriendsView', [])
...
           } else {
-            goodFriends.push(friend);
+            hasPicFriends.push(friend);
           }
         });
-        actualFriends = goodFriends.concat(badFriends);
-        scope.friendGroups = [actualFriends.slice(0,5), actualFriends.slice(5,10)];
+
+        // Expose local friends list on scope.
+        scope.friends = hasPicFriends.concat(noPicFriends);
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/friends/rightColFriendsView.js]: I like the naming improvements and other clarifications you’ve made to this code. In my opinion, the commenting is a little excessive. I think this excerpt from [this post](http://visualstudiomagazine.com/articles/2013/07/26/why-commenting-code-is-still-bad.aspx) fairly summarizes our general attitude toward commenting:

> ... your code is going to communicate to the computer and should also communicate to whoever maintains the code (who may be you, three months from now). To make life easier for the next developer, code should be written and rewritten to be as obvious as possible. What's left for comments to do is explain what the compiler doesn't have access to: why the code is there. Adding comments to explain how your code does its job -- which (I think) some readers were recommending -- is not the solution. It's just adding more cost.
> 
> I will allow that there is one exception: there are times when you're forced to write clever, rather than obvious, code to achieve some goal (usually better performance). In that case a comment about how the code works (really, a kind of apology to the next programmer) is a good idea.

Some of the comments you’ve added are “what” comments, which are a step below “how” comments in utility. I’d recommend removing them all. In my opinion, nothing in here is complex or obscure enough to justify comments. Maybe a single comment stating that we prefer showing users with custom pictures over users with no custom picture, but even that is pretty obvious from the code.

One further clarification that we could make to this code is using `.sort` instead of partitioning into two temporary arrays and then concatenating them back together.

File: kifi-backend/modules/shoebox/angular/src/friends/rightColFriendsView.js [hunk 5]
```diff
@@ -9,23 +9,37 @@ angular.module('kifi.friends.compactFriendsView', [])
     restrict: 'A',
     templateUrl: 'friends/compactFriendsView.tpl.html',
     link: function (scope/*, element, attrs*/) {
+      // Configuration.
+      scope.friendsDisplayLimit = 4;
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/friends/rightColFriendsView.js]: Since this is purely a display concern, consider inlining it into the template. I don’t think the value needs a name.

**Comment by 2is10** [comment]
Minor suggestions. Nice work! Feel free to merge when you’re happy with it.
